nostr-client-plus/src/bin/aggregator.rs

```diff
@@ -403,6 +411,238 @@ async fn run() -> Result<()> {
+        tracing::debug!("Waiting for more results for event id: {}", event_id);
+        return Ok(());
+    }
+    tracing::debug!("All results has been collected for event id: {}", event_id);
+    // max_count count number of results having same max count
+    let mut max_count = 0;
+    // max_answer is number of results having max answer
+    let mut max_answer = 0;
+    let mut answer: String = "".to_string();
+    for result in answers.keys() {
+        let count = answers.get(result).unwrap().to_owned();
+        if count == max_answer {
+            max_count += 1;
+        } else if count > max_answer {
+            max_count = 1;
+            max_answer = count;
+            answer = result.clone();
+        }
+    }
+    let mut tasks_to_update: Vec<AssignerTask> = vec![];
+    if max_count > 1 {
+        tracing::debug!("There are {} results having same count: {}. We need to ask assign same task to more workers.", max_count, max_answer);
+        // Now let's mark all tasks with results as RETRY
+        for task in tasks {
+            if !task.result.is_empty() {
+                tasks_to_update.push(AssignerTask {
+                    worker: task.worker.clone(),
+                    event_id: task.event_id.clone(),
+                    status: AssignerTaskStatus::Retry,
+                    created_at: task.created_at,
+                    expires_at: task.expires_at,
+                    result: task.result.clone(),
+                });
+            }
+        }
+    } else {
+        // Now let's award the winners
+        let mut winners: Vec<Sender> = vec![];
+        for task in tasks {
+            // Safe to unwrap as we have get all results identifiers before
+            let result_identifier = task.get_result_identifier()?.unwrap();
+            if result_identifier == answer {
+                tasks_to_update.push(AssignerTask {
+                    worker: task.worker.clone(),
+                    event_id: task.event_id.clone(),
+                    status: AssignerTaskStatus::Succeeded,
+                    created_at: task.created_at,
+                    expires_at: task.expires_at,
+                    result: task.result.clone(),
+                });
+                winners.push(task.worker.clone());
+            } else if !task.result.is_empty() {
+                tracing::debug!("result_identifier: {}", result_identifier);
+                tracing::debug!("answer: {}", answer);
+                tasks_to_update.push(AssignerTask {
+                    worker: task.worker.clone(),
+                    event_id: task.event_id.clone(),
+                    status: AssignerTaskStatus::Failed,
+                    created_at: task.created_at,
+                    expires_at: task.expires_at,
+                    result: task.result.clone(),
+                });
+            }
+        }
+        tracing::debug!("Winners: {:?}", winners);
+        // job_type 1 as classifier
+        reward(
```

Comment:
There are many things that should be optimized (for example finalization can run in its own tokio task/thread), because so far all this code is executing in a tokio::select branch, slowing down aggregator, but surely `reward` can be run in a spawn task.